torch/_subclasses/fake_tensor.py

```diff
@@ -1129,6 +1135,8 @@ def _output_from_cache_entry(
                 empty.set_(
                     storage, metadata.storage_offset, metadata.shape, metadata.stride
                 )
+        if metadata.storage_bytes == 0:
+            empty.untyped_storage().resize_(0)
```

Comment:
This should fix the issue with fake tensor caching, I deleted the other code I had to temporarily disable fake tensor caching for views